spec.html

```diff
@@ -21416,28 +21894,131 @@ <h1>Static Semantics: ContainsArguments</h1>
+        1. Let _f_ be the active function object.
+        1. Perform ? PrivateSet(_f_.[[PrivateStateKey]], _o_, _value_).
+        1. Return *undefined*.
+      </emu-alg>
+    </emu-clause>
+
     <emu-clause id="sec-runtime-semantics-classfielddefinitionevaluation" type="sdo" aoid="ClassFieldDefinitionEvaluation">
       <h1>Runtime Semantics: ClassFieldDefinitionEvaluation</h1>
 
-      <p>With parameter _homeObject_.</p>
+      <p>With parameters _homeObject_, _decorators_, _metadataList_, and _isStatic_.</p>
 
       <emu-grammar>
         FieldDefinition : ClassElementName Initializer?
       </emu-grammar>
       <emu-alg>
         1. Let _name_ be the result of evaluating |ClassElementName|.
         1. ReturnIfAbrupt(_name_).
+        1. Let _initializers_ be a new empty List.
+        1. Let _finisherInitializers_ be a new empty List.
         1. If |Initializer_opt| is present, then
-          1. Let _formalParameterList_ be an instance of the production <emu-grammar>FormalParameters : [empty]</emu-grammar>.
-          1. Let _scope_ be the LexicalEnvironment of the running execution context.
-          1. Let _privateScope_ be the running execution context's PrivateEnvironment.
-          1. Let _sourceText_ be the empty sequence of Unicode code points.
-          1. Let _initializer_ be ! OrdinaryFunctionCreate(%Function.prototype%, _sourceText_, _formalParameterList_, |Initializer|, ~non-lexical-this~, _scope_, _privateScope_).
-          1. Perform MakeMethod(_initializer_, _homeObject_).
-          1. Set _initializer_.[[ClassFieldInitializerName]] to _name_.
-        1. Else,
-          1. Let _initializer_ be ~empty~.
-        1. Return the ClassFieldDefinition Record { [[Name]]: _name_, [[Initializer]]: _initializer_ }.
+          1. Let _initializer_ be ! CreateFieldInitializerFunction(_name_, _homeObject_, |Initializer|).
+          1. Append _initializer_ to _initializers_.
+        1. For each element _decorator_ of _decorators_, do
+          1. Let _context_ be CreateDecoratorContextObject(*"field"*, _name_, _metadataList_, _finisherInitializers_, _isStatic_).
+          1. Let _initializer_ be ? Call(_decorator_.[[Decorator]], *undefined*, &laquo; *undefined*, _context_ &raquo;).
+          1. If IsCallable(_initializer_) is *true*, append _initializer_ to _initializers_.
+          1. Else if _initializer_ is not *undefined*, throw a *TypeError* exception.
+        1. Return ClassElementDefinition { [[Key]]: _name_, [[Kind]]: ~field~, [[Initializers]]: _initializers_, [[FinisherInitializers]]: _finisherInitializers_ }.
+      </emu-alg>
+      <emu-grammar>
+        FieldDefinition : `accessor` ClassElementName Initializer?
+      </emu-grammar>
+      <emu-alg>
+        1. Let _name_ be the result of evaluating |ClassElementName|.
+        1. ReturnIfAbrupt(_name_).
+        1. Let _privateStateDesc_ be the string-concatenation of *"storage for "* and the StringValue of |ClassElementName|.
```

Comment:
StringValue isn't defined on ClassElementName, and I suspect it shouldn't be. (Currently, StringValue is only defined on Identifiers and similar; ClassElementName is a much wider syntax.)